src/tools/chip-cert/KeyUtils.cpp

```diff
@@ -245,7 +245,7 @@ bool WritePrivateKey(const char * fileName, EVP_PKEY * key, KeyFormat keyFmt)
     uint8_t * keyToWrite   = nullptr;
     uint32_t keyToWriteLen = 0;
     P256SerializedKeypair serializedKeypair;
-    uint32_t chipKeyLen       = sizeof(serializedKeypair);
+    uint32_t chipKeyLen       = serializedKeypair.Capacity();
```

Comment:
This is assuming that Length() and Capacity() are the same, in the non-base64 case.... We should fix that while we are here.

Reply:
The Pull Request has been updated with proper length for the non-base64 case.